pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+	pub struct Pallet<T>(_);
+
+	// The pallet's runtime storage items.
+	// https://substrate.dev/docs/en/knowledgebase/runtime/storage
+	
+	#[pallet::storage]
+	#[pallet::getter(fn invulnerables)]
+	pub type Invulnerables<T: Config>= StorageValue<_, Vec<T::AccountId>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn authors)]
+	pub type Authors<T: Config> = StorageValue<_, Vec<AuthorInfo<T::AccountId, BalanceOf<T>, T::BlockNumber>>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn max_authors)]
+	pub type MaxAuthors<T> = StorageValue<_, u32>; 
+
+	#[pallet::storage]
+	#[pallet::getter(fn authority_bond)]
+	pub type AuthorityBond<T: Config>= StorageValue<_, BalanceOf<T>>;
+
+	
+	#[pallet::event]
+	#[pallet::metadata(T::AccountId = "AccountId")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event documentation should end with an array that provides descriptive names for event
+		/// parameters. [something, who]
+		NewInvulnerables(Vec<T::AccountId>),
+		NewMaxAuthorCount(u32),
+		AuthorAdded(T::AccountId, BalanceOf<T>),
+		AuthorRemoved(T::AccountId)
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		MaxAuthors,
+		Unknown,
+		Permission,
+		AlreadyAuthor,
+		NotAuthor
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		//TODO on init (or finalize) add to aura set at next era 
+		//TODO split half the pot to the author per block
+	}
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsics", which are often compared to transactions.
+	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+
+		#[pallet::weight(10_000)]
+		pub fn set_invulnerables(origin: OriginFor<T>, new: Vec<T::AccountId>) -> DispatchResultWithPostInfo {
+			//TODO chose protective scheme requires message from relay chain 
+			ensure_root(origin)?;
+			<Invulnerables<T>>::put(&new);
+			Self::deposit_event(Event::NewInvulnerables(new));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_max_author_count(origin: OriginFor<T>, max_authors: u32) -> DispatchResultWithPostInfo {
+			//TODO chose protective scheme requires message from relay chain 
+			ensure_root(origin)?;
+			<MaxAuthors<T>>::put(&max_authors);
+			Self::deposit_event(Event::NewMaxAuthorCount(max_authors));
+			Ok(().into())
+
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn author_intent(origin: OriginFor<T>, deposit: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			// lock deposit to start or require min?
+			let who = ensure_signed(origin)?;
+			let length = match <Authors<T>>::decode_len() {
+				Some(len) => len,
+				None => 0
+			};
+			ensure!(length + 1 <= MaxAuthors::<T>::get().ok_or(Error::<T>::MaxAuthors)? as usize, Error::<T>::MaxAuthors); 
```

Comment:
```suggestion
			ensure!(length as u32 < MaxAuthors::<T>::get(), Error::<T>::MaxAuthors); 
```

I've no idea what you were doing here :D  `MaxAuthors` is a raw u32 storage item. If you `::get()` it, it won't return an `Option` or `Result`, but rather just a `u32`. And if it is not set, it will return `0`.

Reply:
Ya so I figured out why this was a challenge for me. With the new pallet writing you have to specify ValueQuery or it will not return a u32 but an option I believe, anyways I need to read into that more, but by adding the value query to the storage item this behaves as expected

Reply:
yeah you need `ValueQuery ` and a default provider.